src/checker.rs

```diff
@@ -1,87 +1,143 @@
+                Some(ref version) => {
+                    bail!(
+                        "Find ckb-cli {} (required {})",
+                        version,
+                        REQUIRED_CKB_CLI_VERSION
+                    );
+                }
+                None => {
+                    bail!(
+                        "Find ckb-cli (unknown version) (required {})",
+                        REQUIRED_CKB_CLI_VERSION
+                    );
+                }
             }
         }
         Ok(())
     }
 
     pub fn print_report(&self) {
         println!("------------------------------");
-        if self.cargo {
-            println!("cargo\tinstalled");
-        } else {
-            println!(
-                "cargo\tnot found - Please install rust (https://www.rust-lang.org/tools/install)"
-            );
-        }
-        if self.docker {
-            println!("docker\tinstalled");
-        } else {
-            println!("docker\tnot found - Please install docker");
+        for (bin_dep, help_message) in [&self.cargo, &self.docker, &self.cross].into_iter().zip(
+            [
+                "Please install rust (https://www.rust-lang.org/tools/install)",
+                "Please install docker",
+                "Please install cross (https://github.com/cross-rs/cross)",
+            ]
+            .into_iter(),
+        ) {
+            if bin_dep.installed {
+                println!("{:10} installed", bin_dep.program);
+            } else {
+                println!("{:10} not found - {}", bin_dep.program, help_message);
+            }
         }
-        if self.ckb_cli.is_some() {
-            match Version::parse_with_prefix("ckb-cli", self.ckb_cli.clone().unwrap()) {
-                Ok(v) if v >= REQUIRED_CKB_CLI_VERSION => {
-                    println!("ckb-cli\tinstalled {}", v);
-                }
-                Ok(v) => {
+
+        let ckb_cli_dep = &self.ckb_cli;
+        if ckb_cli_dep.installed {
+            match &ckb_cli_dep.version {
+                Some(v) => {
                     println!(
-                        "ckb-cli\tinstalled {} (required {})",
-                        v, REQUIRED_CKB_CLI_VERSION
+                        "{:10} installed {} (required {})",
+                        ckb_cli_dep.program, v, REQUIRED_CKB_CLI_VERSION
                     );
                 }
-                Err(_) => {
-                    println!("ckb-cli\tinstalled (unknown version)");
+                None => {
+                    warn!(
+                        "{:10} installed (unknown version) - The deployment feature is disabled",
+                        ckb_cli_dep.program
+                    );
                 }
             }
         } else {
-            println!("ckb-cli\tnot found - The deployment feature is disabled");
+            warn!(
+                "{:10} not found - The deployment feature is disabled",
+                ckb_cli_dep.program
+            );
```

Comment:
Mixing `println!` and log/env_logger doesn't really look good:

```
------------------------------
cargo      installed
docker     installed
cross-util installed
[2023-05-08T04:48:44Z WARN  ckb_capsule::checker] ckb-cli    not found - The deployment feature is disabled
------------------------------
```

And by default only the error level logging is enabled by env_logger, so warn logs are not actually displayed.